irods/test/data_obj_test.py

```diff
@@ -2174,35 +2176,52 @@ def test_touch_operation_does_not_work_when_given_a_collection__525(self):
         with self.assertRaises(ex.InvalidInputArgument):
             user_session.data_objects.touch(home_collection_path)
 
+    def assert_redirect_happens(self, resource, open_opts):
+        name = 'redirect_happens_' + unique_name (my_function_name(), datetime.now())
+        data_path = '{self.coll_path}/{name}'.format(**locals())
+        try:
+            PUT_LOG = self.In_Memory_Stream(always_unicode = True)
+            with helpers.enableLogging(logging.getLogger('irods.manager.data_object_manager'),
+                                       logging.StreamHandler, (PUT_LOG,), level_ = logging.DEBUG):
+                self.sess.data_objects.open(data_path,'w',**open_opts)
```

Comment:
I'm guessing that `auto_close` is set to `True` and that's why this finalizes?

Reply:
Actually, no ... the finally clause here is just for deleting the data object that was opened for purposes of the subtest performed by this helper function.

Reply:
Yes, please make sure it's finalized. Thanks!

Reply:
done